doc/classes/Array.xml

```diff
@@ -544,6 +544,7 @@
 			<param index="0" name="size" type="int" />
 			<description>
 				Resizes the array to contain a different number of elements. If the array size is smaller, elements are cleared, if bigger, new elements are [code]null[/code]. Returns [constant OK] on success, or one of the other [enum Error] values if the operation failed.
+				Calling [method resize] and assigning the new values is faster than appending new elements one by one.
```

Comment:
Note sure if such note is really helpful in the description of `resize`. If someone is not aware of this then they probably use `append`/`push_back` without checking the docs for `resize`. So it might make more sense to add some suggestion in descriptions of `append`/`push_back` instead.

Reply:
Fair point, though in my opinion it's not as practical because reasonably there are 5 places where the note is relevant. I see this note as more of a justification of why one would consider using the method.

I'll change the "appending" wording to "adding", because it's also relevant for inserting and pushing to the front, even if not as much for obvious reason

Reply:
For clarity, I'm referring to: `append()` and `push_back()`, `insert()`, `push_front()`.

Reply:
It's not just for adding though, removing (`erase`, `pop_at`, `pop_back`, `pop_front`) also resizes (and hence can lead to reallocation).

And true, adding notes to all such methods makes little sense. Having it documented at least in on place (`resize`) probably does make sense (if we really assume it's not common knowledge).

Reply:
Personally, I only realized this because I saw performance-sensitive areas of Godot's code utilize it. Maybe it's obvious to people familiar with data structures, but we already address things I consider way more obvious (e.g. how adding elements at the beginning is slower because everything must be reindexed)

Reply:
> Also, as for these destructive methods, I don't think they count because calling resize() first causes a truncation?

To be clear I wasn't saying that just calling `resize()` is enough for all of them. What should be used is case by case. Bulk `pop_back` is like bulk `push_back`/`append`, just a single `resize()` is equivalent/enough. For bulk `insert`, `push_front`, `erase`, `pop_at`, `pop_back` you'd also need some element shifting/manipulation before/after the `resize()`. What exactly depends on what's being done.

Anyway, I'm fine with the current notes. If we'd want to go into details then probably a whole separate manual page in the docs could be written about using/manipulating arrays. :upside_down_face:

Reply:
> Anyway, I'm fine with the current notes. If we'd want to go into details then probably a whole separate manual page in the docs could be written about using/manipulating arrays. 

This is probably a good idea to have, given there are many non-obvious things involving arrays:

- Array vs typed array vs packed array (performance, flexibility, etc)
- Shorthands for initializing packed arrays (implicit casting).
- How to iterate multi-dimensional arrays effectively

This could also just be added to [Data preferences](https://docs.godotengine.org/en/latest/tutorials/best_practices/data_preferences.html).